ledger/complete/wal/compactor.go

```diff
@@ -86,6 +90,11 @@ func (c *Compactor) Run() error {
 		return fmt.Errorf("cannot cleanup checkpoints: %w", err)
 	}
 
+	latest, _ := c.checkpointer.LatestCheckpoint()
```

Comment:
`c.checkpointer.LatestCheckpoint()` triggers a disk access, so we should call it if there are no observers.
Even better we wouldn't check for it at all, as its already called as a part of `createCheckpoints()` call.
Also, if no new checkpoint is created, should we even be emitting new value?

Reply:
thanks, fixed!